### ISSUE TITLE
validate_fastq.sh doesn't catch bzip2 decrompression fail

### ISSUE THREAD

**Issue by anjaf** [nunofonseca/fastq_utils#4]
Hi Nuno, 

A minor bug report. Once in a blue moon we have bzip2 files running through the validator, and the time has come again. ;)

I've noticed that there is a check that is supposed to end the program when the file fails bzip2 decompression 
```
bunzip2   -c $f | gzip -c > $tmp_file
	    if [ $? -ne 0 ]; then
		echo "ERROR: $f: error uncompressing bzip2 file"
exit 2
```
but it didn't run. The error we got happened much later when an empty tmp gzip file was generated/checked. So instead of "error uncompressing bzip2 file" we got "No reads found in xxx.tmp.gz" for the broken bzip2 file.

I've tested the `bunzip | gzip` command with the broken file and it did give exit code 0. So maybe this is something that can be fixed at some point. 

Best wishes,
Anja

**Comment by nunofonseca** [comment]
Hi Anja,

Just to double check, you are referring to the  fastq_validator.sh script and not the validate_fastq.sh?
Assuming yes to the above question, a reason for the check that you mentioned not to  be run is for the file not to have an extension (the type of file is being determined based on the filename extension).  The bzip2 filenames have an extension? 

Cheers,
Nuno

**Comment by nunofonseca** [comment]
Hi Anja, just to ping you regarding the issue.
Cheers.

**Comment by anjaf** [comment]
Hi Nuno,

Sorry I was on holiday. Yes, I mean the fastq_validator.sh script.
In my example the files did have the proper extension (.bz2). 
However, the file was broken. So when I tried to do `bunzip` on it I got the error message: 
> bad magic number (file not created by bzip2)
> You can use the `bzip2recover' program to attempt to recover
> data from undamaged sections of corrupted files."

Now running `bunzip2  -c broken_file.bz2 | gzip -c > tmp_file.fastq.gz`, resulted in an empty temporary fastq.gz file. From looking at the code I would have expected that the script would give me the error "ERROR: broken_file.bz2: error uncompressing bzip2 file", but it didn't, it kept running and reported that there are "no reads found" (in the temp fastq file). 
So I think the error handling can be improved to show the proper reason why validation is failing. 
I hope this is clearer now. Thanks for looking into this. 

Best wishes,
Anja
